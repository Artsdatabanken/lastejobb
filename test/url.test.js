const { url } = require("..");

const input = {
  "AO-KS-FS-JM": {
    kode: "AO-KS-FS-JM",
    tittel: { nb: "Jan Mayen" },
    overordnet: [
      {
        kode: "AO-KS-FS",
        tittel: {
          nb: "Fiskerisone"
        }
      },
      {
        kode: "~",
        tittel: {
          nb: "Usynlig rot"
        }
      }
    ]
  },
  "AO-KS-FS": {
    kode: "AO-KS-FS",
    tittel: {
      nb: "Fiskerisone"
    },
    overordnet: [
      {
        kode: "~",
        tittel: {
          nb: "Usynlig rot"
        }
      }
    ]
  },
  "~": {
    kode: "AO-KS-FS",
    tittel: {
      nb: "Rot"
    },
    overordnet: []
  }
};

// Recursively sort objects
test("createUrl", () => {
  new url(input).assignUrls();
  expect(input).toMatchSnapshot();
});
