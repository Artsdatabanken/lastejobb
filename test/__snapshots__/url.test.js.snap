// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createUrl 1`] = `
Object {
  "AO-KS-FS": Object {
    "kode": "AO-KS-FS",
    "overordnet": Array [
      Object {
        "kode": "~",
        "tittel": Object {
          "nb": "Usynlig rot",
        },
        "url": "Rot",
      },
    ],
    "tittel": Object {
      "nb": "Fiskerisone",
    },
    "url": "Fiskerisone",
  },
  "AO-KS-FS-JM": Object {
    "kode": "AO-KS-FS-JM",
    "overordnet": Array [
      Object {
        "kode": "AO-KS-FS",
        "tittel": Object {
          "nb": "Fiskerisone",
        },
        "url": "Fiskerisone",
      },
      Object {
        "kode": "~",
        "tittel": Object {
          "nb": "Usynlig rot",
        },
        "url": "Rot",
      },
    ],
    "tittel": Object {
      "nb": "Jan Mayen",
    },
    "url": "Fiskerisone/Jan_Mayen",
  },
  "~": Object {
    "kode": "AO-KS-FS",
    "overordnet": Array [],
    "tittel": Object {
      "nb": "Rot",
    },
    "url": "Rot",
  },
}
`;
